crates/bevy_scene/src/dynamic_scene_builder.rs

```diff
@@ -38,31 +57,117 @@ use std::collections::BTreeMap;
 pub struct DynamicSceneBuilder<'w> {
     extracted_resources: BTreeMap<ComponentId, Box<dyn Reflect>>,
     extracted_scene: BTreeMap<Entity, DynamicEntity>,
-    type_registry: AppTypeRegistry,
+    component_filter: SceneFilter,
+    resource_filter: SceneFilter,
     original_world: &'w World,
 }
 
 impl<'w> DynamicSceneBuilder<'w> {
     /// Prepare a builder that will extract entities and their component from the given [`World`].
-    /// All components registered in that world's [`AppTypeRegistry`] resource will be extracted.
     pub fn from_world(world: &'w World) -> Self {
         Self {
             extracted_resources: default(),
             extracted_scene: default(),
-            type_registry: world.resource::<AppTypeRegistry>().clone(),
+            component_filter: SceneFilter::default(),
+            resource_filter: SceneFilter::default(),
             original_world: world,
         }
     }
 
-    /// Prepare a builder that will extract entities and their component from the given [`World`].
-    /// Only components registered in the given [`AppTypeRegistry`] will be extracted.
-    pub fn from_world_with_type_registry(world: &'w World, type_registry: AppTypeRegistry) -> Self {
-        Self {
-            extracted_resources: default(),
-            extracted_scene: default(),
-            type_registry,
-            original_world: world,
-        }
+    /// Specify a custom component [`SceneFilter`] to be used with this builder.
+    pub fn with_filter(&mut self, filter: SceneFilter) -> &mut Self {
+        self.component_filter = filter;
+        self
+    }
+
+    /// Specify a custom resource [`SceneFilter`] to be used with this builder.
+    pub fn with_resource_filter(&mut self, filter: SceneFilter) -> &mut Self {
+        self.resource_filter = filter;
+        self
+    }
+
+    /// Allows the given component type, `T`, to be included in the generated scene.
+    ///
+    /// This method may be called for multiple components.
+    ///
+    /// Please note that this method is mutually exclusive with the [`deny`](Self::deny) method.
+    /// Calling this one will replace the denylist with a new allowlist.
```

Comment:
Hmm: this behavior here is surprising to me. I was expecting this to be additive / incremental, for easy tweaking.

But after some thought, there's no sensible way to have both a list of "allow this" and "deny that": what do you do about unlisted types?

Reply:
Oops I think this is outdated. We did move to an incremental behavior: allowing a type in a denylist simply removes it from that list. I'll update the docs to match!